src/native/fs/fs_napi.cpp

```diff
@@ -1385,6 +1386,18 @@ fs_napi(Napi::Env env, Napi::Object exports)
 {
     auto exports_fs = Napi::Object::New(env);
 
+    const char* env_p = std::getenv("GPFS_DL_PATH");
+    if (env_p != NULL) {
+        LOG("FS::GPFS GPFS_DL_PATH=" << env_p);
+        uv_lib_t *lib = (uv_lib_t*) malloc(sizeof(uv_lib_t));
+        if (uv_dlopen(env_p, lib)) {
+            PANIC("Error: %s\n" << uv_dlerror(lib));
+        }
+        if (uv_dlsym(lib, "gpfs_fcntl", (void **) &gpfs_fcntl)) {
```

Comment:
OK, read this -
https://stackoverflow.com/questions/20321802/i-am-about-to-use-dlopen-to-open-shared-object-do-i-need-to-include-correspon

What I understand is that technically the exported functions from headers are not "used" when using dlsym because the addresses are not resolved during link phase. However if we use constants or structures then the header is still useful of course. The question is if we're doing the right thing when assigning to the gpfs_fcntl name here - does it really change the name from the header? Did you see an example that does this same pattern (update the function pointed by the header)?

Reply:
Adding @eshelmarc

Reply:
I guess you used this example -
http://docs.libuv.org/en/v1.x/guide/utilities.html#loading-libraries

Reply:
I wonder what actually happens with the current code.
Which pointer is updated by dlsym?

Reply:
@guymguym '[Which pointer is updated by dlsym?](https://github.com/noobaa/noobaa-core/pull/6933#discussion_r845147085)' I might be missing something however, [this](https://github.com/jackyalbo/noobaa-core/blob/ff417caf12679d72ae0ffa603afd080473ef1b8d/src/native/fs/fs_napi.cpp#L38) is the definition of a pointer to `gpfs_fcntl ` function **variable** and not a declaration. dlsym writes the pointer to the requested function to that variable.

Reply:
do you want to fail if library is not found or execute the functions in a way that don't need GPFS functions.

Reply:
> do you want to fail if library is not found or execute the functions in a way that don't need GPFS functions.

@eshelmarc I don't think it's necessary - this is why we condition it on the existence of the `GPFS_DL_PATH` env, so if you want to load it without the library you just omit that env, but if you provide it - you want to make sure it is loaded and used.

Reply:
> guymguym 'Which pointer is updated by dlsym?' I might be missing something however, this is the definition of a pointer to function **variable** and not a declaration. dlsym writes the pointer to the requested function ( `gpfs_fcntl` ) to that variable.

@baum Yes I understan that but was asking because the same symbol name is also defined as a function in gpfs headers. Nevermind though, Jacky already answered that the code I commented on was not even compiling.